Lib/defcon/objects/font.py

```diff
@@ -1382,11 +1408,92 @@ def _convertToFormatVersion1RoboFabData(self, libCopy):
                         finalValues.append([])
                     finalValues[-1].append(value)
                 hintData[libKey] = finalValues
-        for key, value in hintData.items():
+        for key, value in list(hintData.items()):
             if value is None:
                 del hintData[key]
         libCopy["org.robofab.postScriptHintData"] = hintData
 
+    # -----------------------------
+    # Serialization/Deserialization
+    # -----------------------------
+
+    def getDataForSerialization(self, **kwargs):
+        from functools import partial
+
+        simple_get = partial(getattr, self)
+        serialize = lambda item: item.getDataForSerialization()
+        serialized_get = lambda key: serialize(simple_get(key))
+
+        getters = (
+            ('_ufoFormatVersion', simple_get),
+            ('_kerningGroupConversionRenameMaps', simple_get),
+            # _path ? => setting path may change the behavior when deserializing a lot!
+            # also path should be set by the caller, e.g. when saving the font
+            # otherwise, path should be deserialized as the very last item,
+            # because otherwise the font object will try to load a lot data
+            # from disk, when deserializing.
+            ('data', serialized_get),
+            ('features', serialized_get),
+            ('groups', serialized_get),
+            ('images', serialized_get),
+            ('info',  serialized_get),
+            ('kerning', serialized_get),
+            ('layers',  serialized_get),
+            ('lib', serialized_get)
+        )
+
+        return self._serialize(getters, **kwargs)
+
+    def setDataFromSerialization(self, data):
+        from functools import partial
+
+        set_attr = partial(setattr, self) # key, data
+
+        def single_update(key, data):
```

Comment:
I'm not opposed to nested functions, but I agree that I'd rather them be private methods or helper functions if possible. I find nested functions to be hard to read so my standard formatting practice is to avoid them.